src/scaler/spreadscaler/link.rs

```diff
@@ -1,54 +1,55 @@
-use std::{
-    collections::{BTreeMap, HashMap},
-    hash::{Hash, Hasher},
-};
+use std::hash::{Hash, Hasher};
 
 use anyhow::Result;
 use async_trait::async_trait;
-use tokio::sync::{OnceCell, RwLock};
-use tracing::{instrument, trace};
+use tokio::sync::RwLock;
+use tracing::instrument;
 
 use crate::{
     commands::{Command, DeleteLinkdef, PutLinkdef},
     events::{
-        Event, LinkdefDeleted, LinkdefSet, ProviderHealthCheckPassed, ProviderHealthCheckStatus,
+        Event, LinkdefDeleted, LinkdefSet, ProviderHealthCheckInfo, ProviderHealthCheckPassed,
+        ProviderHealthCheckStatus,
     },
     model::TraitProperty,
     scaler::Scaler,
     server::StatusInfo,
-    storage::{Actor, Provider, ReadStore},
+    storage::ReadStore,
     workers::LinkSource,
-    APP_SPEC_ANNOTATION, DEFAULT_LINK_NAME,
 };
 
 pub const LINK_SCALER_TYPE: &str = "linkdefscaler";
 
 /// Config for a LinkSpreadConfig
 pub struct LinkScalerConfig {
-    /// OCI, Bindle, or File reference for the actor to link
-    actor_reference: String,
-    /// OCI, Bindle, or File reference for the provider to link
-    provider_reference: String,
-    /// Contract ID the provider implements
-    provider_contract_id: String,
-    /// Contract ID the provider implements
-    provider_link_name: String,
+    /// Component identifier for the source of the link
+    pub source_id: String,
+    /// Target identifier or group for the link
+    pub target: String,
+    /// WIT Namespace for the link
+    pub wit_namespace: String,
+    /// WIT Package for the link
+    pub wit_package: String,
+    /// WIT Interfaces for the link
+    pub wit_interfaces: Vec<String>,
+    /// Name of the link
+    pub name: String,
     /// Lattice ID the Link is configured for
-    lattice_id: String,
+    pub lattice_id: String,
     /// The name of the wadm model this SpreadScaler is under
-    model_name: String,
-    /// Values to attach to this linkdef
-    values: HashMap<String, String>,
+    pub model_name: String,
+    /// List of named configurations for the source of this link
+    pub source_config: Vec<String>,
+    /// List of named configurations for the target of this link
+    pub target_config: Vec<String>,
 }
 
 /// The LinkSpreadScaler ensures that link configuration exists on a specified lattice.
 pub struct LinkScaler<S, L> {
     pub config: LinkScalerConfig,
+    // TODO: Reenable once we figure out https://github.com/wasmCloud/wadm/issues/123
```

Comment:
I honestly wonder if we don't even need this anymore. With the new links, you really have to be trying hard if you are linking the exact same IDs and providers across multiple manifests

Reply:
If we scope config to manifests, and generate IDs for components, there's no way that a user could hit this without doing it themselves so I think we can handle it simply by using an exponential backoff model.